builtin/providers/dns/test_check_attr_string_array.go

```diff
@@ -27,24 +27,27 @@ func testCheckAttrStringArray(name, key string, value []string) r.TestCheckFunc
 			return fmt.Errorf("Attributes not found for %s", attrKey)
 		}
 
-		got, _ := strconv.Atoi(count)
-		if got != len(value) {
+		gotCount, _ := strconv.Atoi(count)
+		if gotCount != len(value) {
 			return fmt.Errorf("Mismatch array count for %s: got %s, wanted %d", key, count, len(value))
 		}
 
-		for i, want := range value {
+	Next:
```

Comment:
Aye, should have clarified, sorry! 
Functionally, this looks good to me, however I feel it's slightly over complex for a simple test function. Easy to miss (as I did above), the labeled statement on first read. That being said, however, it's not critical enough to request another change on it 😄 
Would gladly accept a future PR to refactor this test function a bit, though.